pkg/collector/corechecks/network-devices/cisco-sdwan/payload/devices.go

```diff
@@ -0,0 +1,92 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package payload implement processing of Cisco SD-WAN api responses
+package payload
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/network-devices/cisco-sdwan/client"
+	devicemetadata "github.com/DataDog/datadog-agent/pkg/networkdevice/metadata"
+)
+
+// TimeNow useful for mocking
+var TimeNow = time.Now
+
+// ProcessDevices process devices API payloads to build metadata, tags and uptimes
+func ProcessDevices(namespace string, devices []client.Device) (devicesMetadata []devicemetadata.DeviceMetadata, deviceTags map[string][]string, uptimes map[string]float64) {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Seems like the logic to get metadata, uptimes, tags is different. I'm wondering if it should be separated in sub functions for better isolation. No strong opinion for this

Reply:
Also it could be good to test this function (or the sub functions if split)

Reply:
🤔 Mmmh I guess this is to catch errors and make sure all return arrays have the same size? Another possibility could be to use a custom type if we really need to group these three together later

Reply:
Yeah I've iterated a lot between the two 😅 Main reason I ended up grouping everything together is to iterate only one time on devices / interfaces, which honestly might be me over-optimising things.
I think separating makes sense, easier to read and to test, updating this!

Reply:
Addressed in 6d6e42c